lib/utils.js

```diff
@@ -68,6 +55,11 @@ const log = (message, level) => {
   const time = `${padStart(dateObject.getHours(), 2)}:${padStart(dateObject.getMinutes(), 2)}:${padStart(dateObject.getSeconds(), 2)}`
   const tzOffset = dateObject.getTimezoneOffset() * -100 / 60
   const timezone = `${(tzOffset > 0 ? '+' : '')}${padStart(tzOffset, 4)}`
+  // Properly handle Error objects.  The client may have additional meta information to log
+  if (typeof message === 'object' && message.constructor && message.constructor.name === 'Error') {
```

Comment:
@darinspivey, can you show me as an example where this kind of thing can happen in the newer changes? I mean, where is `message` gonna be an `object`?

Reply:
Just as an extra context: these `console` statements generates the logs which are getting into `/var/log/logdna-agent.log` from which they get streamed into and parsed on `LogDNA` where we parse `datetime`, `timezone`, and `level` info

Reply:
While testing, I got a nock error, and all that came out was "there was a problem" from the client.  [Here is the place](https://github.com/logdna/logdna-agent/blob/e7ae9b24b7c0f0f9acb996402058d8e20b1f92e1/lib/logger-client.js#L32) where this change will benefit us.  It will now look like this:

```
2020-09-17 12:50:18 -0400 [error] { Error: An error occured while sending logs to the cloud.
    at process.nextTick (/Users/dspivey/github/logdna-agent/node_modules/@logdna/logger/lib/logger.js:719:25)
    at process._tickCallback (internal/process/next_tick.js:61:11)
  meta:
   { actual:
      'Nock: No match for request {\n  "method": "POST",\n  "url": "https://logs.logdna.com/logs/agent?now=1600361418306&hostname=MyHostMachine&mac=... [snip]\\"}]}"\n}',
     code: 'ERR_NOCK_NO_MATCH',
     firstLine: 'Here is my line1',
     lastLine: 'Here is my line 2',
     retrying: false,
     attempts: 1 } }
```

Reply:
I see @darinspivey but if possible why don't we change how we log the error [here](https://github.com/logdna/logdna-agent/blob/e7ae9b24b7c0f0f9acb996402058d8e20b1f92e1/lib/logger-client.js#L32) instead of touching this `log` method since only newer change `(logger-client)` addition created necessity to add this handling but I don't think it's gonna be used anywhere else?

Reply:
afaict this isn't necessary other than to use `console.error` instead, is that right? without this, the full error would still be passed to `console.log` now with the other changes

Reply:
No extra concerns here - not gonna block this and it's easy to update whenever needed and/or requested in the future

Reply:
Sorry, @mdeltito , forgot to answer you.  Yes, the big thing here was using `console.error`.  With `console.log`, the full error was being truncated for some reason.  We *really* need to use pino, but our pino config is not public, however that shouldn't stop us from doing it.